typing/typemod.ml

```diff
@@ -1830,29 +1849,33 @@ and type_module_aux ~alias sttn funct_body anchor env smod =
       let funct =
         type_module (sttn && path <> None) funct_body None env sfunct in
       begin match Env.scrape_alias env funct.mod_type with
-        Mty_functor(param, mty_param, mty_res) as mty_functor ->
-          let generative, mty_param =
-            (mty_param = None, Btype.default_mty mty_param) in
-          if generative then begin
-            if sarg.pmod_desc <> Pmod_structure [] then
-              raise (Error (sfunct.pmod_loc, env, Apply_generative));
-            if funct_body && Mtype.contains_type env funct.mod_type then
-              raise (Error (smod.pmod_loc, env, Not_allowed_in_functor_body));
-          end;
+      | Mty_functor (None, mty_res) ->
+          if sarg.pmod_desc <> Pmod_structure [] then
+            raise (Error (sfunct.pmod_loc, env, Apply_generative));
+          if funct_body && Mtype.contains_type env funct.mod_type then
+            raise (Error (smod.pmod_loc, env, Not_allowed_in_functor_body));
+          (* FIXME: is this even needed? *)
+          check_well_formed_module env smod.pmod_loc
+            "the signature of this functor application" mty_res;
```

Comment:
@garrigue (or @lpw25): I don't think the signature of an generative functor needs to be rechecked on applications. Can you confirm?

Reply:
Ping @garrigue / @lpw25 .

Reply:
I agree it shouldn't need re-checking.

Reply:
Thanks, I pushed a commit removing that (and another similar) call.